contracts/counter/README.md

```diff
@@ -0,0 +1,255 @@
+
+```rust,ignore
+    // Previous
+    let balance: BalanceResponse = cosmwasm_smart_query("my_cw20_address", cw20::QueryMsg::Balance{
+        address: "my_address".to_string(),
+    })?;
+
+    // Now
+    let balance = my_cw20.balances("my_address".to_string())?;
+```
+
+Better right ?
+
+### Additional attributes
+
+#### Payable
+Let's see an example for executing a message (from a money market for instance)
+
+```rust,ignore
+    money_market.deposit_stable()?;
+```
+
+This money market however only knows how much you deposit into it by looking at the funds you send along with the transaction. Cw-orchestrator prevents that by default. However, to allow depositing funds with a transaction, you can add the `#[payable]` attribute on your enum variant like so : 
+
+```rust, ignore
+    #[derive(ExecuteFns)]
+    enum ExecuteMsg{
+        UpdateConfig{
+            config_field: String
+        },
+        #[payable]
+        DepositStable{}
+        ...
+    }
+```
+
+Be defining this attribute, you can now use : 
+```rust,ignore
+    money_market.deposit_stable(coins(456, "ujunox"))?;
+```
+
+#### fn_name
+Be careful what you name your enum variants, you could stumble upon one of the reserved keywords of cw-orch. If this happens, you can use the fn_name attribute to rename a generated function.
+
+```rust,ignore
+
+    ExecuteMsg{
+        #[fn_name("proxy_execute")]
+        Execute{
+            msg: CosmoMsg
+        }
+    }
+
+    // Will error because the execute function is reserved for contract execution. Will not even compile actually
+    money_market.execute(message_to_execute_via_a_proxy)?;
+```
+
+```rust,ignore
+    ExecuteMsg{
+        #[fn_name("proxy_execute")]
+        Execute{
+            msg: CosmoMsg
+        }
+    }
+    // This works smoothly !
+    money_market.proxy_execute(message_to_execute_via_a_proxy)?;
+```
+
+This is also true for query functions.
+
+
+#### impl_into
+For more details about this attribute, [see the full documentation](https://orchestrator.abstract.money/single_contract/interfaces.html#impl_into-attribute)
+
+
+
+
+## Other cw-orch traits
+
+### Deploy
+
+cw-orchestrator defines the [Deploy](../../cw-orch/src/deploy.rs) trait that allows developers to create an interface that allows themselves or other developers to : 
+1. Simply deploy their codebase on a test chain or elsewhere. [See it live here](https://github.com/AbstractSDK/abstract/blob/main/framework/scripts/src/bin/full_deploy.rs#L29).
+2. Provide bindings (addresses, code_ids), for other developers to integrate with the platform. [See it live here](https://github.com/AbstractSDK/abstract/blob/main/modules/contracts/apps/croncat/examples/deploy.rs#L32).
+
+This is even more powerful than simply exposing the single contract interfaces, because it allows full customization of the deployment procedure. It also allows shipping other custom methods for simplifying the life of the integrator.
```

Comment:
```suggestion
This is more powerful than simply exposing the single contract interfaces, because it allows full customization of the deployment procedure. It also allows shipping other custom methods for simplifying the life of the integrator.
```